R/lazyframe-frame.R

```diff
@@ -254,33 +519,98 @@ lazyframe__cast <- function(..., .strict = TRUE) {
+#'
+#' # Filter by comparing two columns against each other
+#' lf$filter(pl$col("foo") == pl$col("bar"))$collect()
+#' lf$filter(pl$col("foo") != pl$col("bar"))$collect()
+#'
+#' # Notice how the row with null values is filtered out$ In order to keep the
+#' # rows with nulls, use:
+#' lf$filter(pl$col("foo")$ne_missing(pl$col("bar")))$collect()
 lazyframe__filter <- function(...) {
   parse_predicates_constraints_into_expression(...) |>
     self$`_ldf`$filter() |>
     wrap()
 }
 
+#' Sort the LazyFrame by the given columns
+#'
+#' @param ... <[`dynamic-dots`][rlang::dyn-dots]> Column(s) to sort by. Can be
+#' character values indicating column names or Expr(s).
+#' @param descending Sort in descending order. When sorting by multiple
+#' columns, this can be specified per column by passing a logical vector.
+#' @param nulls_last Place null values last. When sorting by multiple
+#' columns, this can be specified per column by passing a logical vector.
+#' @param maintain_order Whether the order should be maintained if elements are
+#' equal. If `TRUE`, streaming is not possible and performance might be worse
+#' since this requires a stable search.
+#' @param multithreaded Sort using multiple threads.
+#'
+#' @inherit as_polars_lf return
+#' @examples
+#' lf <- pl$LazyFrame(
+#'   a = c(1, 2, NA, 4),
+#'   b = c(6, 5, 4, 3),
+#'   c = c("a", "c", "b", "a")
+#' )
+#'
+#' # Pass a single column name to sort by that column.
+#' lf$sort("a")$collect()
+#'
+#' # Sorting by expressions is also supported
+#' lf$sort(pl$col("a") + pl$col("b") * 2, nulls_last = TRUE)$collect()
+#'
+#' # Sort by multiple columns by passing a vector of columns
+#' lf$sort(c("c", "a"), descending = TRUE)$collect()
+#'
+#' # Or use positional arguments to sort by multiple columns in the same way
+#' lf$sort("c", "a", descending = c(FALSE, TRUE))$collect()
 lazyframe__sort <- function(
     ...,
     descending = FALSE,
     nulls_last = FALSE,
     multithreaded = TRUE,
     maintain_order = FALSE) {
-  wrap({
-    check_dots_unnamed()
+  check_dots_unnamed()
 
-    by <- parse_into_list_of_expressions(...)
-    descending <- extend_bool(descending, length(by), "descending", "...")
-    nulls_last <- extend_bool(nulls_last, length(by), "nulls_last", "...")
-
-    self$`_ldf`$sort_by_exprs(
-      by,
-      descending = descending,
-      nulls_last = nulls_last,
-      multithreaded = multithreaded,
-      maintain_order = maintain_order
-    )
-  })
+  by <- parse_into_list_of_expressions(...)
+  if (length(by) == 0) {
+    abort("`...` must contain at least one element.")
+  }
+  descending <- extend_bool(descending, length(by), "descending", "...")
+  nulls_last <- extend_bool(nulls_last, length(by), "nulls_last", "...")
+
+  self$`_ldf`$sort_by_exprs(
+    by,
+    descending = descending,
+    nulls_last = nulls_last,
+    multithreaded = multithreaded,
+    maintain_order = maintain_order
+  ) |> wrap()
```

Comment:
Why?

Reply:
I got confused at some point on whether `wrap()` should include the whole code or just the call to `self$`_ldf`$<fn>()`, I'll revert this.